docs/docs/migration-guide.mdx

```diff
@@ -126,6 +126,152 @@ confidences. This change can affect the maximum confidence predicted by the
+
+```yaml
+slots:
+  is_existing_customer:
+    type: bool
+    mappings:
+    - type: custom
+      action: action_verify_customer_status
+```
+
+You can also use [slot validation actions](./slot-validation-actions.mdx) to define custom validation of slots with
+predefined mappings or custom extraction and validation of slots with custom mappings.
+
+:::note
+The `required_slots` of a form has been converted in a list of slot names because slot mappings are now relocated
+from `forms` to `slots` in the domain.
+:::
+
+#### Automatic migration from 2.0 domain format to the 3.0 format
+
+The only data file that has changed in format is the domain file.
+To migrate automatically to the 3.0 domain format, you can run the following command:
+
+```bash
+rasa data migrate -d DOMAIN --out OUT_PATH
+```
+
+In addition to creating a valid 3.0 domain in the indicated out path, this command will automatically backup your
+original domain file(s) in a file labeled `original_domain.yml` or `original_domain` directory if a directory was
+provided instead.
+
+To maintain the behavior of forms in the 2.0 format, all migrated slot mappings will include mapping conditions for
+each form. This can be changed manually according to your use case.
+See the docs on [mapping conditions](./domain.mdx#mapping-conditions) for more information.
+
+#### Manually migrating from 2.0 domain format to the 3.0 format
+
+Each slot in the `slots` section of the domain will need a new key `mappings`.
+This key is a list of mappings moved from forms, while the `required_slots` field collapses to a list of slot names.
+
+Let's consider the following 2.0 domain file:
+
+```yaml
+entities:
+ - cuisine
+ - number
+ slots:
+   cuisine:
+     type: text
+   num_people:
+     type: float
+   outdoor_seating:
+     type: bool
+ forms:
+   restaurant_form:
+     required_slots:
+         cuisine:
+           - type: from_entity
+             entity: cuisine
+         num_people:
+           - type: from_entity
+             entity: number
+         outdoor_seating:
+           - type: from_intent
+             intent: affirm
+             value: true
+           - type: from_intent
+             intent: deny
+             value: false
+```
+
+The initial result of this migration could look like this:
+
+```yaml
+entities:
+ - cuisine
+ - number
+ slots:
+   cuisine:
+     type: text
+     mappings:
+     - type: from_entity
+       entity: cuisine
+   num_people:
+     type: float
+     mappings:
+     - type: from_entity
+       entity: number
+   outdoor_seating:
+     type: bool
+     mappings:
+     - type: from_intent
+       intent: affirm
+       value: true
+     - type: from_intent
+       intent: deny
+       value: false
+ forms:
+   restaurant_form:
+     required_slots:
+	 - cuisine
+     - num_people
+     - outdoor_seating
+```
+
+In case you'd prefer certain slots to be filled only within the context of a form, you can add mapping conditions like so:
```

Comment:
```suggestion
For slots that should be filled only in the context of a form, add mapping conditions to specify which form(s) should be active. Adding `conditions` is required preserve the behavior of slot mappings from 2.0, since without them the mappings will be applied on each user turn regardless of whether a form is active or not.
```